plugins/modules/df_service.py

```diff
@@ -305,57 +313,65 @@ def process(self):
             field=['status', 'state'], state=self.cdpy.sdk.STARTED_STATES,
             delay=self.delay, timeout=self.timeout
         )
 
     def _disable_df(self):
         # Attempt clean Disable, which also ensures we have tried at least once before we do a forced removal
         if self.target['status']['state'] in self.cdpy.sdk.REMOVABLE_STATES:
-            self.service = self.cdpy.df.disable_environment(
-                env_crn=self.env_crn,
-                persist=self.persist
+            self.service = self.cdpy.df.disable_service(
+                df_crn=self.df_crn,
+                persist=self.persist,
+                terminate=self.terminate
             )
+        else:
+            self.module.warn("Attempting to disable DataFlow Service but state %s not in Removable States %s"
+                             % (self.target['status']['state'], self.cdpy.sdk.REMOVABLE_STATES))
         if self.wait:
             # Wait for Clean Disable, if possible
             self.service = self.cdpy.sdk.wait_for_state(
-                describe_func=self.cdpy.df.describe_environment, params=dict(env_crn=self.env_crn),
+                describe_func=self.cdpy.df.describe_service, params=dict(df_crn=self.df_crn),
                 field=['status', 'state'],
                 state=self.cdpy.sdk.STOPPED_STATES + self.cdpy.sdk.REMOVABLE_STATES + [None],
                 delay=self.delay, timeout=self.timeout, ignore_failures=True
             )
         else:
-            self.service = self.cdpy.df.describe_environment(env_crn=self.name)
+            self.service = self.cdpy.df.describe_service(df_crn=self.df_crn)
         # Check disable result against need for further forced delete action, in case it didn't work first time around
         if self.service is not None:
             if self.service['status']['state'] in self.cdpy.sdk.REMOVABLE_STATES:
                 if self.force:
-                    self.service = self.cdpy.df.delete_environment(
-                        env_crn=self.env_crn
+                    self.service = self.cdpy.df.reset_service(
+                        df_crn=self.df_crn
                     )
                 else:
                     self.module.fail_json(msg="DF Service Disable failed and Force delete not requested")
             if self.wait:
                 self.service = self.cdpy.sdk.wait_for_state(
-                    describe_func=self.cdpy.df.describe_environment, params=dict(env_crn=self.env_crn),
+                    describe_func=self.cdpy.df.describe_service, params=dict(df_crn=self.df_crn),
                     field=None,  # This time we require removal or declare failure
                     delay=self.delay, timeout=self.timeout
                 )
             else:
-                self.service = self.cdpy.df.describe_environment(env_crn=self.name)
+                self.service = self.cdpy.df.describe_service(df_crn=self.df_crn)
 
 
 def main():
     module = AnsibleModule(
```

Comment:
Consistent feedback across PRs :) 

Can we add tags?

Reply:
Added as a feature request: https://github.com/cloudera-labs/cloudera.cloud/issues/37